### ISSUE TITLE
Question about reproduction

### ISSUE THREAD

**Issue by AceCoooool** [xiaoyufenfei/LEDNet#3]
I want to implement your paper, but have several questions: 
1. is there missing first convolution before downsampling in the paper ? (from channel 3 to channel 32)
2. Did you pretrained you encoder in imagenet or just scratch ? (And whould you mind to tell the training epochs you use ?)
3. What kind of weight init strategy you used?
4. The training image size

Here are my implementaion : https://github.com/AceCoooool/LEDNet 
But there is still some gap to the paper

**Comment by xiaoyufenfei** [comment]
1. Inspired by ENet, the downsampling module uses parallel branches. You can refer to the settings in ENet. After Concat, there are also Bn and ReLU.
2. There is no problem with the number of channels from 3 to 32, which is implemented by the downsampling module.
3. If your GPU resources are sufficient, you can pre-train it on ImageNet, and the results may be better. It provides more categories and plays a regular role.
4. When training, Cityscapes size is 1024x512. If your GPU resources are sufficient, it is recommended that Batch_size be set larger, such as greater than 6.
5. Dropout should be set for each branch of each SS-nbt building module. LEDNet is inspired by ENet. You can refer to the Dropout setting in ENet. For example, set a smaller drop_rate in the shallow layer and a larger drop_rate in the middle layer. Sorry, The paper is not clearly described. In addition, all the bias should be set to True when training from the beginning. I hope this helps you.

**Comment by AceCoooool** [comment]
Thank you so much !

**Comment by AceCoooool** [comment]
I am sorry, I change setting as you said. However, the performance is still bad (drop-out with 0.01 and 0.1---last 8 SS-nbt). Is there some other tricks?

**Comment by xiaoyufenfei** [comment]
Have you changed it? For example, all eight SS-nbt modules in the middle have drop_rate set to 0.3, the remaining SS-nbt are set to 0.03, and other settings, sorry

**Comment by xiaoyufenfei** [comment]
@AceCoooool 
In my settings:
# In DownSampling you also can use Dropout after Bn
class DownSampling(nn.Module):
    def __init__(self, in_channel, out_channel):
        super(DownSampling, self).__init__()
        self.conv = nn.Conv2d(in_channel, out_channel-in_channel, 3, stride=2, padding=1,bias=True)
        self.pool = nn.MaxPool2d(2, stride=2)
        self.bn = nn.BatchNorm2d(out_channel)
        self.relu = nn.ReLu(inplace=True)

    def forward(self, x):
        output = torch.cat([self.conv(x),self.pool(x)], dim=1)
        output = self.bn(output)
        output = self.relu(output)
        return output

## SS-nbt module is slight different，for example in one branch
Conv3x1 -> ReLu ->Conv1x3 -> Bn -> ReLu -> Conv3x1 -> ReLu ->Conv1x3 -> Bn -> Dropout

## May be you set all inplace=True ,you can make a better result

## In my experiments, all eight SS-nbt modules in the middle have drop_rate set to 0.3, the other SS-nbt module are set to 0.03，I find different Dropouts are influential in my experiments. may  different dropout set can get a better result
